src/VisualStudio/CSharp/Impl/Debugging/LocationInfoGetter.cs

```diff
@@ -43,49 +28,93 @@ internal static class LocationInfoGetter
             {
-                SeparatedSyntaxList<VariableDeclaratorSyntax> variableDeclarators = ((BaseFieldDeclarationSyntax)memberDecl).Declaration.Variables;
+                SeparatedSyntaxList<VariableDeclaratorSyntax> variableDeclarators = ((BaseFieldDeclarationSyntax)memberDeclaration).Declaration.Variables;
 
                 foreach (var declarator in variableDeclarators)
                 {
                     if (declarator.FullSpan.Contains(token.FullSpan))
                     {
-                        memberDecl = declarator;
+                        fieldDeclarator = declarator;
                         break;
                     }
                 }
 
-                if (memberDecl == null)
+                if (fieldDeclarator == null)
                 {
-                    memberDecl = variableDeclarators.Count > 0 ? variableDeclarators[0] : null;
+                    fieldDeclarator = variableDeclarators.Count > 0 ? variableDeclarators[0] : null;
                 }
             }
 
-            if (memberDecl != null)
+            if (memberDeclaration != null)
             {
-                var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
-                var memberSymbol = semanticModel.GetDeclaredSymbol(memberDecl, cancellationToken);
+                const char dotTokenText = '.';
+                const string missingInformationPlaceholder = "?";
 
-                if (memberSymbol != null)
+                var builder = s_stringBuilderPool.AllocateAndClear();
+                // simple name
+                var nameToken = fieldDeclarator?.Identifier ?? memberDeclaration.GetNameToken();
+                if (nameToken.IsMissing)
+                {
+                    builder.Append(missingInformationPlaceholder);
+                }
+                else if (nameToken == default(SyntaxToken))
+                {
+                    Debug.Assert(memberDeclaration.Kind() == SyntaxKind.ConversionOperatorDeclaration);
+                    builder.Append((memberDeclaration as ConversionOperatorDeclarationSyntax)?.Type);
+                }
+                else
+                { 
+                    if (memberDeclaration.Kind() == SyntaxKind.DestructorDeclaration)
+                    {
+                        builder.Append('~');
+                    }
+                    builder.Append(nameToken.Text);
+                }
+                // parameter list (if any)
+                builder.Append(memberDeclaration.GetParameterList());
+                // containing type(s)
+                var containingDeclaration = memberDeclaration.GetAncestor<TypeDeclarationSyntax>();
+                SyntaxNode lastContainer = memberDeclaration;
+                while (containingDeclaration != null)
+                {
+                    builder.Insert(0, dotTokenText);
```

Comment:
I've moved types/namespaces up front, but I'm still inserting them (to avoid having to collect them as I walk up through the ancestors or reverse the list).